### PR TITLE
fix(tag-list): No longer shows 0 more

### PR DESCRIPTION
**Pull request by rowa-audil**

### CODE REVIEW

**Comment by rowa-audil** [comment]
@ffriedl89 
IMO a test wouldn't make sense in that case since I already test that and the problem only occurred when the change detection didn't glance over the tags again (for some reason)

File: components/tag/src/tag-list/tag-list.html [hunk 1]
```diff
@@ -6,7 +6,10 @@
       *ngIf="_showAllTags"
     ></ng-template>
   </div>
-  <div class="dt-tag-list-more-btn" *ngIf="!_isOneLine && !_showAllTags">
+  <div
+    class="dt-tag-list-more-btn"
+    *ngIf="!_isOneLine && !_showAllTags && _hiddenTagCount > 0"
```
Comment (lukasholzer) [inline components/tag/src/tag-list/tag-list.html]: NIT: I would create a simple function that evaluates this – keeping templates simple!

**Comment by thomaspink** [comment]
When review is done, please reword and rebase to make it clear for the consumer what this fix tries to solve.

**Comment by tomheller** [comment]
Fixes #235

File: components/tag/src/tag-list/tag-list.ts [hunk 2]
```diff
@@ -196,6 +200,17 @@ export class DtTagList implements AfterContentInit, OnDestroy {
     });
     this._tagAddSubscriptions = [];
   }
+
+  /**
+   * @internal evaluates whether to display the x more button
+   */
+  _toDisplayMoreButton(): void {
+    if (!this._isOneLine && !this._showAllTags && this._hiddenTagCount > 0) {
```
Comment (lukasholzer) [inline components/tag/src/tag-list/tag-list.ts]: according to our style guide please use a conditional operator for that and furthermore use the function directly in the template no need to store in a variable


```typescript
_showDisplayMoreButton(): boolean {
  return !this._isOneLine && !this._showAllTags && this._hiddenTagCount > 0
}
```

File: components/tag/src/tag-list/tag-list.ts [hunk 3]
```diff
@@ -96,6 +96,9 @@ export class DtTagList implements AfterContentInit, OnDestroy {
   /** @internal Whether user wants all tags to be shown */
   _showAllTags = false;
 
+  /** @internal Whether the x more button should be displayed */
+  _displayMoreBtn = false;
```
Comment (lukasholzer) [inline components/tag/src/tag-list/tag-list.ts]: no need to store this in a variable use the function directly in the template

File: components/tag/src/tag-list/tag-list.html [hunk 4]
```diff
@@ -6,7 +6,7 @@
       *ngIf="_showAllTags"
     ></ng-template>
   </div>
-  <div class="dt-tag-list-more-btn" *ngIf="!_isOneLine && !_showAllTags">
+  <div class="dt-tag-list-more-btn" *ngIf="_displayMoreBtn">
```
Comment (lukasholzer) [inline components/tag/src/tag-list/tag-list.html]: call the function here directly 
```suggestion
  <div class="dt-tag-list-more-btn" *ngIf="_showDisplayMoreButton()">
```

File: apps/components-e2e/src/components/tag-list/tag-list.e2e.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Selector } from 'testcafe';
+
+const tagList = Selector('#tagList');
+
+fixture.only('TagList').page('http://localhost:4200/tag/tag-list');
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.e2e.ts]: you forgot the .only please remove

File: apps/components-e2e/src/components/tag-list/tag-list.e2e.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+test("should not display '0 More...'", async (testController: TestController) => {
+  await testController.resizeWindow(700, 500);
+  await testController.expect(tagList.textContent).notContains('0 More...');
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.e2e.ts]: chain the expect to the resize and add a `.wait(300)` between those to wait a little bit after resizing

File: apps/components-e2e/src/components/tag-list/tag-list.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!-- <div class="tag-list"> -->
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.html]: please remove comment

File: apps/components-e2e/src/components/tag-list/tag-list.html [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+  <dt-tag>Beta</dt-tag>
+  <dt-tag>Gamma</dt-tag>
+  <dt-tag>Delta</dt-tag>
+  <dt-tag>Epsilon</dt-tag>
+  <dt-tag>Zeta</dt-tag>
+  <dt-tag>Eta</dt-tag>
+  <dt-tag>Theta</dt-tag>
+  <dt-tag-add></dt-tag-add>
+</dt-tag-list>
+<!-- </div> -->
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.html]: please remove comment

File: apps/components-e2e/src/components/tag-list/tag-list.scss [hunk 9]
```diff
@@ -0,0 +1,5 @@
+.tag-list {
+  width: 700px;
+  height: auto;
+  background: red;
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.scss]: I think you can remove this :)

File: apps/components-e2e/src/components/tag-list/tag-list.scss [hunk 10]
```diff
@@ -0,0 +1,5 @@
+.tag-list {
+  width: 700px;
```
Comment (lukasholzer) [inline apps/components-e2e/src/components/tag-list/tag-list.scss]: please remove that you resize the window

**Comment by rowa-audil** [comment]
Squashed commits, waiting for review.

**Comment by rowa-audil** [comment]
Had a copypasta error :'(

**Pull request merged**